File: 0995/solution.py
from collections import deque


class Solution:
    def minKBitFlips(self, nums: list[int], k: int) -> int:
        flips = deque()
        res = 0
        for i in range(len(nums)):
            if flips and flips[0] < i:
                flips.popleft()
            curr = (nums[i] + len(flips)) % 2
            if curr == 0:
                if i > len(nums) - k:
                    return -1
                flips.append(i + k - 1)
                res += 1
        return res

File: 0995/solution.ts
function minKBitFlips(nums: number[], k: number): number {
  const flips: Array<number> = [];
  let res = 0;
  for (let i = 0; i < nums.length; i++) {
    if (flips.length && flips[0] < i) {
      flips.shift();
    }
    let curr = (nums[i] + flips.length) % 2;
    if (curr == 0) {
      if (i > nums.length - k) {
        return -1;
      }
      res++;
      flips.push(i + k - 1);
    }
  }
  return res;
}
